### PR TITLE
SNOW-870360: Integrate with code cov

### PR DESCRIPTION
**Pull request by sfc-gh-dprzybysz**
### Description
- report code coverage in lcov format next to text
- add codecov uploader of gathered code coverage with nyc
- add codecov badge

### Checklist
- [x] Format code according to the existing code style (run `npm run lint:check -- CHANGED_FILES` and fix problems in changed code)
- [x] Create tests which fail without the change (if possible)
- [x] Make all tests (unit and integration) pass (`npm run test:unit` and `npm run test:integration`)
- [x] Extend the README / documentation and ensure is properly displayed (if necessary)
- [x] Provide JIRA issue id (if possible) or GitHub issue id in commit message

### CODE REVIEW

File: .github/workflows/build-test.yml [hunk 1]
```diff
@@ -69,10 +69,11 @@ jobs:
                 PARAMETERS_SECRET: ${{ secrets.PARAMETERS_SECRET }}
                 CLOUD_PROVIDER: ${{ matrix.cloud }}
               run: /usr/local/bin/bash ./ci/test_mac.sh
-            - uses: actions/upload-artifact@master
+            - name: Upload lcov coverage directory
+              uses: actions/upload-artifact@master
```
Comment (sfc-gh-pmotacki) [inline .github/workflows/build-test.yml:73]: maybe here  you can use concrete version of upload-artifact instead of master

Comment (sfc-gh-dprzybysz) [reply .github/workflows/build-test.yml:73]: set to v1

File: ci/container/test_component.sh [hunk 2]
```diff
@@ -88,7 +88,7 @@ python3 $THIS_DIR/hang_webserver.py 12345 > hang_webserver.out 2>&1 &
 if [[ "$SHOULD_GENERATE_COVERAGE_REPORT" -eq "1" && "$CLOUD_PROVIDER" == "AWS" ]];
   then
     MOCHA_CMD=(
-       "npx" "nyc" "mocha" "--exit" "--timeout" "$TIMEOUT" "--recursive" "--full-trace"
+       "npx" "nyc" "--reporter=lcov" "--reporter=text" "mocha" "--exit" "--timeout" "$TIMEOUT" "--recursive" "--full-trace"
```
Comment (sfc-gh-pmotacki) [inline ci/container/test_component.sh:91]: It isn't need to set output directory as "coverage"?

Comment (sfc-gh-dprzybysz) [reply ci/container/test_component.sh:91]: No, it's lcov default

**Review by sfc-gh-pmotacki — APPROVED** [review]
LGTM

**Pull request merged**